esp-hal-common/src/timer.rs

```diff
@@ -17,6 +22,70 @@ pub enum Error {
     AlarmInactive,
 }
 
+pub struct TimerGroup<T>
+where
+    T: TimerGroupInstance,
+{
+    pub timer0: Timer<Timer0<T>>,
+    #[cfg(not(feature = "esp32c3"))]
+    pub timer1: Timer<Timer1<T>>,
+    pub wdt: Wdt<T>,
+}
+
+pub trait TimerGroupInstance {
+    fn register_block() -> *const RegisterBlock;
+}
+
+impl TimerGroupInstance for TIMG0 {
+    #[inline(always)]
+    fn register_block() -> *const RegisterBlock {
+        crate::pac::TIMG0::PTR
+    }
+}
+
+impl TimerGroupInstance for TIMG1 {
+    #[inline(always)]
+    fn register_block() -> *const RegisterBlock {
+        crate::pac::TIMG1::PTR
+    }
+}
+
+impl<T> TimerGroup<T>
+where
+    T: TimerGroupInstance,
+{
+    pub fn new(_timer_group: T, clocks: &Clocks) -> Self {
+        let timer0 = Timer::new(
+            Timer0 {
+                phantom: PhantomData::default(),
+            },
+            clocks.apb_clock,
+        );
+
+        #[cfg(not(feature = "esp32c3"))]
+        let timer1 = Timer::new(
+            Timer1 {
+                phantom: PhantomData::default(),
+            },
+            clocks.apb_clock,
+        );
+
+        let wdt = Wdt::new();
+
+        #[cfg(not(feature = "esp32c3"))]
+        let timer_group = Self {
+            timer0,
+            timer1,
+            wdt,
+        };
+
+        #[cfg(feature = "esp32c3")]
+        let timer_group = Self { timer0, wdt };
+
+        timer_group
```

Comment:
```suggestion
        Self {
            timer0,
            #[cfg(not(feature = "esp32c3"))]
            timer1,
            wdt,
        }
```

This can be simplified.